docs/Architecture.md

```diff
@@ -1,42 +1,55 @@
-Maestro v10 (aka Next) is a complete rewrite of the service, focusing on improving code maintainability, knowing what is being done on the rooms/schedulers, and changing the state management.
+Maestro Next is a set of modules that together deliver all features that it is built for. Internally they are all part of the same code base and could be run by giving the right arguments to your command line, or giving the same arguments to your docker container entrypoint. The execution process could be changed by giving correct arguments as environment variables (a proper list will be documented in the correct section).
 
-The main focus of the refactor was to take Maestro's strengths and areas that could be improved. After brainstorming, we've decided that a simpler and more concrete model was necessary to maximize those areas.
+Maestro is composed of Management API, Rooms API, Operation Execution Worker, Runtime Watcher Worker, and Metrics Reporter Worker. Each one with its own responsibilities and is broken enough in a way to avoid mixing the execution process. Each module was thought to avoid parallel problems and to give the client more visibility about which **Operations** are being executed and their respective status.
```

Comment:
```suggestion
Maestro is composed of **Management API**, **Rooms API**, **Operation Execution Worker**, **Runtime Watcher Worker**, and **Metrics Reporter Worker.** 

Each module has its responsibilities and is divided apart in a way to avoid mixing the execution process. Each module was thought to avoid parallel problems and to give the client more visibility about which [**Operations**](Operations.md) are being executed and their respective status.
```